paymentsheet/src/main/java/com/stripe/android/paymentsheet/viewmodels/BaseSheetViewModel.kt

```diff
@@ -414,9 +415,8 @@ internal abstract class BaseSheetViewModel(
         if (selection is PaymentSelection.New) {
             newPaymentSelection = selection
         }
-
         savedStateHandle[SAVE_SELECTION] = selection
-
+        clearErrorMessages()
```

Comment:
Will the `clearErrorMessage()` not be called when a new value of `selection` is emitted (see the fragment)? Why do we need this here?